services/helper/sessions.js

```diff
@@ -1,64 +1,703 @@
+                });
+            }
+            const response = await sessionAttendesData.unEnrollFromSession(sessionId, userId);
+
+            if (response === 'USER_NOT_ENROLLED') {
+                return common.failureResponse({
+                    message: apiResponses.USER_NOT_ENROLLED,
+                    statusCode: httpStatusCode.bad_request,
+                    responseCode: 'CLIENT_ERROR'
+                });
+            }
+
+            const templateData = await notificationTemplateData.findOneEmailTemplate(process.env.MENTEE_SESSION_CANCELLATION_EMAIL_TEMPLATE);
+
+            if (templateData) {
+                // Push successfull unenrollment to session in kafka
+                const payload = {
+                    type: 'email',
+                    email: {
+                        to: email,
+                        subject: templateData.subject,
+                        body: utils.composeEmailBody(templateData.body, {
+                            name,
+                            sessionTitle: session.title,
+                            mentorName: session.mentorName
+                        })
+                    }
+                };
+
+                await kafkaCommunication.pushEmailToKafka(payload);
+            }
+
+            return common.successResponse({
+                statusCode: httpStatusCode.accepted,
+                message: apiResponses.USER_UNENROLLED_SUCCESSFULLY,
+            });
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    static async verifyMentor(id) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                let options = {
+                    "headers": {
+                        'Content-Type': "application/json",
+                        "internal_access_token": process.env.INTERNAL_ACCESS_TOKEN
+                    }
+                };
+
+                let apiUrl = apiBaseUrl + apiEndpoints.VERIFY_MENTOR + "?userId=" + id;
+                try {
+                    request.post(apiUrl, options, callback);
+
+                    function callback(err, data) {
+                        if (err) {
+                            return reject({
+                                message: apiResponses.USER_SERVICE_DOWN
+                            });
+                        } else {
+                            data.body = JSON.parse(data.body);
+                            if (data.body.result && data.body.result.isAMentor) {
+                                return resolve(true);
+                            } else {
+                                return resolve(false);
+
+                            }
+                        }
+                    }
+                } catch (error) {
+                    reject(error);
+                }
+
+            } catch (error) {
+                reject(error);
+            }
+        });
+    }
+
+    static getAllAccountsDetail(userIds) {
```

Comment:
@rakeshSgr  How does getAllAccountsDetail relate to sessions ?

Reply:
@aman-g202 please check this

Reply:
To find email data of all eligible mentees when session gets cancelled or rescheduled

Reply:
Moved in attendees helper